src/sage/matroids/circuits_matroid.pyx

```diff
@@ -0,0 +1,613 @@
+r"""
+Circuits matroids
+
+Matroids are characterized by a list of circuits, which are minimal dependent
+sets. The CircuitsMatroid class implements matroids using this information as
+data.
+
+A ``CircuitsMatroid`` can be created from another matroid or from a list of
+circuits. For a full description of allowed inputs, see
+:class:`below <sage.matroids.circuits_matroid.CircuitsMatroid>`. It is
+recommended to use the :func:`Matroid() <sage.matroids.constructor.Matroid>`
+function for a more flexible construction of a ``CircuitsMatroid``. For direct
+access to the ``CircuitsMatroid`` constructor, run::
+
+    sage: from sage.matroids.circuits_matroid import CircuitsMatroid
+
+AUTHORS:
+
+- Giorgos Mousa (2023-12-23): initial version
+
+TESTS::
+
+    sage: from sage.matroids.circuits_matroid import CircuitsMatroid
+    sage: M = CircuitsMatroid(matroids.catalog.Fano())
+    sage: TestSuite(M).run()
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2023 Giorgos Mousa <gmousa@proton.me>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.structure.richcmp cimport rich_to_bool, richcmp
+from sage.matroids.matroid cimport Matroid
+from sage.matroids.set_system cimport SetSystem
+from cpython.object cimport Py_EQ, Py_NE
+
+
+cdef class CircuitsMatroid(Matroid):
+    r"""
+    INPUT:
+
+    - ``M`` -- a matroid (default: ``None``)
+    - ``groundset`` -- a list (default: ``None``); the groundset of the matroid
+    - ``circuits`` -- a list (default: ``None``); the collection of circuits of
+      the matroid
+
+    OUTPUT:
+
+    - If the input is a matroid ``M``, return a ``CircuitsMatroid`` instance
+      representing ``M``.
+    - Otherwise, return a ``CircuitsMatroid`` instance based on the
+      ``groundset`` and ``circuits``.
```

Comment:
```suggestion
```
There is no point as the output is clearly this class. This isn't uniform and very, very in Sage (and likely should be removed in other places).